libraries/AC_Fence/AC_Fence.cpp

```diff
@@ -103,8 +131,34 @@ void AC_Fence::enable(bool value)
     }
     _enabled = value;
     if (!value) {
-        clear_breach(AC_FENCE_TYPE_ALT_MAX | AC_FENCE_TYPE_CIRCLE | AC_FENCE_TYPE_POLYGON);
+        clear_breach(AC_FENCE_TYPE_ALT_MIN | AC_FENCE_TYPE_ALT_MAX | AC_FENCE_TYPE_CIRCLE | AC_FENCE_TYPE_POLYGON);
+    } else {
+        // turn the floor back on if it had been off
+        floor_enable(true);
+    }
+}
+
+/// enable/disable fence floor only
+void AC_Fence::floor_enable(bool value)
+{
+    if (_floor_enabled && !value) {
+        //! TODO: Log?
+    } else if(!_floor_enabled && value) {
+        //! TODO: Log?
+    }
+    _floor_enabled = value;
+    if (!value) {
+        clear_breach(AC_FENCE_TYPE_ALT_MIN);
+    }
+}
+
+bool AC_Fence::present() const
+{
+    // A fence is present if it is enabled and there are points present
```

Comment:
We could have many inclusion / exclusion circles which Plane should now honour - and `present` ought to be true then!

Reply:
So, a fences' presence is only tied to whether one is uploaded?

ie. We can have a fence (inclusion/exclusion/both) uploaded, but not enabled.

Reply:
I've updated this to check inclusion/exclusion fences for circle and poly being present.